### PR TITLE
Clean up gzdev calls and use docker cache via ADD for gzdev changes

### PR DESCRIPTION
**Pull request by j-rivero**
The PR clean up most of the gzdev code that deal with possible changes in gzdev configuraton in different parts of the Dockerfile:  The mechanism used is to leverage the caching actions to the Docker layer caching using the `ADD https://api.github.com/repos/gazebo-tooling/gzdev/git/refs/heads/$branch version.json` that should invalidate the cache on changes done to the branch inside gzdev (affecting code or configurations). The current implementation will invalidate all the cache on any change to the gzdev branch.

The PR also changes the installation of gzdev to be under the /root directory.

### CODE REVIEW

File: jenkins-scripts/docker/lib/docker_generate_dockerfile.bash [hunk 1]
```diff
@@ -40,48 +40,50 @@ fi
...
 if [[ "${DISTRO}" == 'bionic' || "${DISTRO}" == 'focal' ]]; then
   KEYSERVER="hkps://pgp.surf.nl"
 fi
 
 dockerfile_install_gzdev_repos()
 {
-cat >> Dockerfile << DELIM_OSRF_REPO_GIT
-RUN rm -fr ${GZDEV_DIR}
-RUN git clone https://github.com/gazebo-tooling/gzdev -b ${GZDEV_BRANCH} ${GZDEV_DIR}
-RUN if [ -n $GZDEV_TRY_BRANCH ]; then \
```
Comment (j-rivero) [inline jenkins-scripts/docker/lib/docker_generate_dockerfile.bash:59]: To support also docker caching the `GZDEV_TRY_BRANCH` this check has been taken out of the Dockerfile and a new ADD to check the GZDEV_TRY_BRANCH was added in adding to the two git commands.

**Comment by j-rivero** [comment]
I've been executing it several times in https://build.osrfoundation.org/job/_test_gz_math-ci-gz-math7-noble-amd64/ playing with changes in gzdev and try branch. Seemed to me to work as expected.

**Comment by j-rivero** [comment]
Going to merge this one before it stays for too long out of my radar. [Build Status](https://build.osrfoundation.org/job/gz_tools-ci-pr_any-noble-amd64/3/) another test build

File: jenkins-scripts/docker/lib/docker_generate_dockerfile.bash [hunk 2]
```diff
@@ -40,48 +40,50 @@ fi
...
-        git -C ${GZDEV_DIR} fetch origin $GZDEV_TRY_BRANCH || true; \
-        git -C ${GZDEV_DIR} checkout $GZDEV_TRY_BRANCH || true; \
-    fi || true
+cat >> Dockerfile << DELIM_OSRF_REPO_GIT_1
+ADD https://api.github.com/repos/gazebo-tooling/gzdev/git/refs/heads/$GZDEV_BRANCH version.json
```
Comment (scpeters) [inline jenkins-scripts/docker/lib/docker_generate_dockerfile.bash:57]: we sometimes see 504 "failed to get" errors from this line

what if this script used `curl` with a retry loop, with a fallback to `git` if needed, to download `version.json` and then `ADD` the locally downloaded file to the `Dockerfile`?

**Pull request merged**